### DISCUSSION TITLE
Is there a way to know the queue time on self-hosted runner

### DISCUSSION THREAD

**Discussion by Anthony-Tubi** [community/community#38035] [Apps, API and Webhooks]
After I migrated to the self-hosted runner, queuing often occurs. I want to count the queuing time of recent jobs. Is there any API that supports it?

**Comment by ghostinhershell** [comment]
Hi there @Anthony-Tubi 👋 and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

- To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
- To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776)
- You can further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121)
- Finally, you can also participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

Cheers!

**Comment by snehalDH** [comment]
Hi @Anthony-Tubi, did you discover anything related to job queue time ? Even I am trying to get queue time for jobs and workflows.

**Reply by Anthony-Tubi** [reply]
@snehalDH  The self-hosted runner is supported by our backend team in k8s, there are several reasons that might cause queues
- Lots of IO in your job, which will share total IO resources
- Waiting for k8s pod expansion
- Insufficient k8s pod resources
You can check them to locate the problem

To address getting queue time, I write a script via GitHub REST API and calculate the difference between creation time and start time, it's about queue time I guess.

**Reply by FWBlueWizard** [reply]
This is a great question would be great to know as well how many jobs are waiting to run and with what labels they're using. Diff that with what runners are available and which are actively in progress we can build our own queue of jobs waiting to be executed and how long they've been waiting..

**Comment by soufiiyane** [comment]
u can use timestamp tracking on a self-hosted runner, use a script to record the start timestamp in a logger file or a database, that will do the trick !

**Reply by FWBlueWizard** [reply]
Yeah this is what I was thinking. I was contacted by GH team to get sold enterprise. Asked if they had queues for self-hosted. Short answer was still no and they just pointed me to the github-hosted runners.. which isn't a solution for self-hosted >.<
